### ISSUE TITLE
Blazor WASM not passing claims to Web API 

### ISSUE THREAD

**Issue by eistudios** [dotnet/aspnetcore#25147]
### Describe the bug
I have configured a custom claim using Okta that is attached in my identity token. Blazor WASM when using the AddHttpClient method does not add those claims values to the token. In my example this is included in the groups claim and is a json array of groups.  The requirement in the code below checks for that policy. 

This requirement check works on the Blazor WASM side. But fails on the Web API side. Looking at the JWT I captured from the header that claim is not included.

### To Reproduce
I am using the boiler plate code. 

```
            builder.Services.AddOidcAuthentication(options =>
            {
                builder.Configuration.Bind("Okta", options.ProviderOptions);
                options.ProviderOptions.ResponseType = "code";
                options.ProviderOptions.DefaultScopes.Add("profile");
            });
            builder.Services.AddSingleton<IAuthorizationHandler, HasSiteAdmin>();
            builder.Services.AddAuthorizationCore(options =>
            {
                options.AddPolicy("SiteAdmin", policy =>
                   policy.Requirements.Add(new IsSiteAdmin()));
            });


            builder.Services.AddHttpClient("KeeperLife.UI.ServerAPI", client => client.BaseAddress = new Uri(builder.HostEnvironment.BaseAddress))
                .AddHttpMessageHandler<BaseAddressAuthorizationMessageHandler>();

            // Supply HttpClient instances that include access tokens when making requests to the server project
            builder.Services.AddScoped(sp => sp.GetRequiredService<IHttpClientFactory>().CreateClient("KeeperLife.UI.ServerAPI"));
```

### Further technical details
.NET Core SDK (reflecting any global.json):
 Version:   3.1.400
 Commit:    035fb2aa2f

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18362
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.400\

Host (useful for support):
  Version: 3.1.6
  Commit:  3acd9b0cd1

.NET Core SDKs installed:
  2.1.102 [C:\Program Files\dotnet\sdk]
  2.1.104 [C:\Program Files\dotnet\sdk]
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.801 [C:\Program Files\dotnet\sdk]
  2.2.106 [C:\Program Files\dotnet\sdk]
  3.0.100 [C:\Program Files\dotnet\sdk]
  3.1.300 [C:\Program Files\dotnet\sdk]
  3.1.400 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.0.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 3.1.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download

Visual Studio Enterprise 2019 version 16.7.0

**Comment by eistudios** [comment]
This is what I previoulsy had on the server side. 
```
services.AddAuthentication(options =>
            {
                  options.DefaultScheme = JwtBearerDefaults.AuthenticationScheme;
            }).AddJwtBearer(options =>
            {
                options.Authority = oktaSettings.Authority;
                options.Audience = "api://default";
                options.RequireHttpsMetadata = false;
                options.TokenValidationParameters = new TokenValidationParameters
                {
                    IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(oktaSettings.ClientSecret)),
                    RequireSignedTokens = true,
                    ClockSkew = TimeSpan.FromMinutes(5),

                    RequireExpirationTime = true,
                    ValidateLifetime = true,

                    ValidateAudience = true,
                    ValidAudience = "api://default",

                    ValidateIssuer = true,
                    ValidIssuer = oktaSettings.Authority

                };
            });
```
I first attempted to add the lines above with out changing the Add IdentityServer and that did not work. I secondly attempted to add the AddIdentityServerJwt call but it want not found. I added the nuget package 

Microsoft.AspNetcore.Authentication.JwtBearer but that did not work. 

I am assuming that you meant on the Try to user at server-side meant you want it on the API side of the project and not on the blazor side. 

When looking at the bearer token that is passed from the client to the backend. I am not seeing the data in the bearer token attached to the call. 

This is the bearer token I was able to grab from the call directly from the browser. 


```
{
  "ver": 1,
  "jti": "<REMOVED>",
  "iss": "<REMOVED>",
  "aud": "<REMOVED>",
  "iat": <REMOVED>,
  "exp": <REMOVED>,
  "cid": "<REMOVED>",
  "uid": "<REMOVED>",
  "scp": [
    "profile",
    "groups",
    "openid"
  ],
  "sub": "<REMOVED>"
}
```

It is not passing the groups section.

**Comment by mkArtakMSFT** [comment]
@eistudios can you try to follow the guidance provided here: https://docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/hosted-with-identity-server?view=aspnetcore-3.1&tabs=visual-studio#api-authorization-options

**Comment by eistudios** [comment]
I am receiving a token from okta from WASM. This is an identity token, which I am 100% sure contains the groups claims. When this is being passed from WASM to server the claim is not included. I have validated this by capturing the JWT during transit from browser to back end. 

The recommended advice appears to be setup on the server side.

**Comment by eistudios** [comment]
So inside the Blazor app on one of the authenticated pages I added 

```
        var tokenResult = await TokenProvider.RequestAccessToken();

        tokenResult.TryGetToken(out var token);
        bearerToken = token.Value;
```

Taking this token and using jwt.io decryptor I see the claim for groups is not in there as well.

**Comment by eistudios** [comment]
All right updating this bad boy for the next person who runs across this. I am making some assumptions here based off of what I am seeing and maybe some one who understands this better can validate, but I wanted to share. When Blazor first authenticates against Okta it does so using an identity token.  In my setup of the custom group export in Okta I set the config for the custom claim to be attached to the identity token. This works fine and lets me use policies to validate authorization based off of those claims.  Now this is where I start to guess a little. When it goes through and sends the token to the back end web api it is obtaining an access token from Okta and sending it to the API. Since I did not configure the custom claim on the access token it was missing when it sent it to the web api. Adding this configuration to the server to add this claim to the access token, corrected this error and caused the custom claim to be passed to the API.
